program/src/processor.rs

```diff
@@ -100,6 +88,8 @@ fn get_total_stake_checked(
         return Err(ProgramError::UninitializedAccount);
     }
 
+    // TODO: How to check the config account is the correct one?
+
```

Comment:
Flagging: Since the PDA won't work without the validator key, we need another way to check that the config account is the right config for the stake account. One idea is we modify `Stake` state to store a pointer to its corresponding config.

Reply:
Pass the vote account as instruction data, verify it's both the vote account stored in the `Stake` account as well as the vote account for the PDA seed for `Config`?

Reply:
I've got this set up locally. It obviously works fine, but it's a bit funky. This makes validators provide stake, vote, and stake authority when casting a vote. Are we cool with this? Or should I make the vote account the signer again, and just use the stake for the vote PDA, etc. as I did in my most recent commit?

Reply:
We definitely shouldn't make the vote account a signer, since it's not a guarantee that validators have that keypair still. I totally forgot that the validator is stored in the stake account, why don't we just use that to make sure that the stake PDA is correct? That derivation uses the stake config, which is validated by the governance config, so we should be safe, right?

Reply:
@buffalojoec I am following the logic that @joncinque mentioned in the `HarvestStakeRewards`, since the instruction does not expect the validator vote account.

https://github.com/paladin-bladesmith/stake-program/blob/2a2bdc09c432d558db4448cd4efa6d262323555d/program/src/processor/harvest_stake_rewards.rs#L60-L76

Reply:
Done!